api/resolvers/paidAction.js

```diff
@@ -1,6 +1,44 @@
 import { retryPaidAction } from '../paidAction'
+import { USER_ID } from '@/lib/constants'
+
+function paidActionType (actionType) {
+  switch (actionType) {
+    case 'ITEM_CREATE':
+    case 'ITEM_UPDATE':
+      return 'ItemPaidAction'
+    case 'ZAP':
+    case 'DOWN_ZAP':
+      return 'ItemActPaidAction'
+    case 'TERRITORY_CREATE':
+    case 'TERRITORY_UPDATE':
+    case 'TERRITORY_BILLING':
+    case 'TERRITORY_UNARCHIVE':
+      return 'SubPaidAction'
+    case 'DONATE':
+      return 'DonatePaidAction'
+    case 'POLL_VOTE':
+      return 'PollVotePaidAction'
+    default:
+      throw new Error('Unknown action type')
+  }
+}
 
 export default {
+  Query: {
+    paidAction: async (parent, { invoiceId }, { models, me }) => {
+      const invoice = await models.invoice.findUnique({ where: { id: invoiceId, userId: me?.id ?? USER_ID.anon } })
```

Comment:
I can't think of why we'd need to protect those invoices:

- the sender's lightning privacy is great
- the only data "leaked" that otherwise isn't is the time the invoice was created/paid/etc